runtime/src/main/java/org/capnproto/ReaderArena.java

```diff
@@ -45,11 +45,13 @@ public SegmentReader tryGetSegment(int id) {
     }
 
     @Override
-    public final void checkReadLimit(int numBytes) {
-        if (numBytes > limit) {
+    public final void checkReadLimit(int numWords) {
```

Comment:
Oh, wow, this definitely should be `numWords`, not `numBytes` as it was. Good catch!